tests/Framework/TestCaseTest.php

```diff
@@ -671,4 +671,20 @@ private function getAutoreferencedArray()
             ]
         ];
     }
+
+    public function testProvidingOfSelfReferencingArray()
```

Comment:
what's the difference between autoreferenced array and selfreferencing one?
no need for testing `$arr[] = &arr` here as well, as it's tested in previous test